### PR TITLE
os/memstore: Fix memory leak

### PR DESCRIPTION
**Pull request by aclamk**
There was a permanent memory leak, objects were never deleted.
ObjectRef x(new Bufferlist()) created RefCountedObject with nref=2 (one for new, one for ref).
ObjectRef x = make_ref<Bufferlist>() just makes one.

Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by aclamk** [comment]
This leak can be observed by: 
```valgrind --leak-check=full ./bin/unittest_pglog```

**Review by athanatos — APPROVED** [review]

File: src/os/memstore/MemStore.cc [hunk 1]
```diff
@@ -1820,5 +1820,5 @@ int MemStore::PageSetObject::truncate(uint64_t size)
 MemStore::ObjectRef MemStore::Collection::create_object() const {
   if (use_page_set)
     return ceph::make_ref<PageSetObject>(cct->_conf->memstore_page_size);
-  return new BufferlistObject();
+  return make_ref<BufferlistObject>();
```
[the remark this answers is not in the document]
Comment (aclamk) [reply src/os/memstore/MemStore.cc:1823]: Is there any other `make_ref` then the one in `ceph` namespace?
If so, I will update the PR to clarify that.

**Comment by aclamk** [comment]
jenkins test make check

**Comment by aclamk** [comment]
jenkins test api

**Comment by idryomov** [comment]
A later tracker ticket: https://tracker.ceph.com/issues/58701

**Pull request merged**